### PR TITLE
statistics: make `HotThresholdRatio` configurable

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Ref #5692

### What is changed and how does it work?

The current threshold for hotspot statistics, when the number of hot regions is too large, we will use the value of top60 as the threshold. This top60 will count the hot regions within three minutes.

For clusters with drastic load changes, hotspot statistics will consume a lot of memory. As a temporary means, we need to be able to configure the threshold of statistics
image

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mock/mockcluster/config.go [hunk 1]
```diff
@@ -111,6 +111,11 @@ func (mc *Cluster) SetHotRegionCacheHitsThreshold(v int) {
 	mc.updateScheduleConfig(func(s *config.ScheduleConfig) { s.HotRegionCacheHitsThreshold = uint64(v) })
 }
 
+// SetHotThresholdRatio updates the HotThresholdRatio configuration.
+func (mc *Cluster) SetHotThresholdRatio(v int) {
```
Comment (rleungx) [inline pkg/mock/mockcluster/config.go:115]: used?

File: server/config/config.go [hunk 2]
```diff
@@ -776,6 +776,9 @@ type ScheduleConfig struct {
 
 	// EnableWitness is the option to enable using witness
 	EnableWitness bool `toml:"enable-witness" json:"enable-witness,string"`
+
+	// HotThresholdRatio is used to calculate hot thresholds
+	HotThresholdRatio float64 `toml:"hot-threshold-ratio" json:"hot-threshold-ratio,omitempty"`
```
Comment (rleungx) [inline server/config/config.go:781]: Do we need to validate the value of this config?

Comment (lhy1024) [reply server/config/config.go:781]: add non-negative check

**Comment by lhy1024** [comment]
This PR won't be released in v6.5.0.

**Pull request closed without merging**